docs/sphinx/_ext/sphinx-mdx-builder/sphinxcontrib/mdxbuilder/writers/mdx.py

```diff
@@ -168,6 +168,77 @@ def __init__(self, document: nodes.document, builder: "MdxBuilder") -> None:
             ord(">"): "&gt;",
         }
 
+    def _unwrap_function_object(self, obj):
+        """Attempt to unwrap function-like objects to get the underlying callable.
+
+        This method handles various common patterns used by decorators and frameworks
+        that wrap functions in custom objects while preserving access to the original function.
+
+        Args:
+            obj: The object to unwrap
+
+        Returns:
+            The unwrapped function if found, otherwise the original object
+        """
+        # Common patterns for accessing wrapped functions
+        function_attributes = [
+            # Standard functools.wraps pattern (already handled above via __wrapped__)
+            "__wrapped__",
+            # Common patterns in various frameworks:
+            "func",  # Used by many decorators
+            "function",  # Alternative naming
+            "__func__",  # Method objects
+            "fget",  # Property objects
+            "__call__",  # Callable objects (last resort)
+            # Framework-specific patterns (generic names to avoid coupling):
+            "logger_fn",  # Dagster LoggerDefinition
+            "op_fn",  # Dagster OpDefinition
+            "resource_fn",  # Dagster ResourceDefinition
+            "sensor_fn",  # Dagster SensorDefinition
+            "schedule_fn",  # Dagster ScheduleDefinition
+            "hook_fn",  # Dagster HookDefinition
+            "decorated",  # Generic decorated function access
+            "inner",  # Generic inner function access
+            "wrapped",  # Alternative to __wrapped__
+            "_func",  # Private function storage
+            "_fn",  # Private function storage (short form)
+            "callback",  # Callback-style wrappers
+            "handler",  # Handler-style wrappers
+        ]
```

Comment:
Hold-up, I want to ensure this is generic and has no hard-coding for Dagster specific code.